core/compress_api.php

```diff
@@ -83,6 +83,13 @@ function compress_handler_is_enabled() {
  * @access public
  */
 function compress_start_handler() {
+	# Do not start compress handler if we got any output so far, as this
+	# denotes that an error has occurred. Enabling compression in this case
+	# will likely cause a Content Encoding Error when displaying the page.
+	if( ob_get_length() ) {
```

Comment:
I believe @atrol's point about performance was due to my using `ob_get_contents()`, when the buffer's contents is not actually needed. 

> would @ob_start() fix the error surfacing?

No, the error occurs because we have some output prior to starting output buffering with the compression handler, it generates mixed contents that the browser can't interpret.

>  Would setting a flag in trigger_error() and checking it here help?

IMO it's a better approach to avoid triggering the error in the first place - disabling the compression handler in the (rare) case where initializing MantisBT core generates output efficiently resolves the problem IMO.